content/posts/web-frameworks.md

```diff
@@ -0,0 +1,132 @@
+
+## The outcome
+
+Here's what I landed on:
+
+| Topic | Rails | Svelte | HTMX/FastAPI |
+|---------|:-----:|:------:|:------:|
+| Lang.      | -- | -- | <i class="fa-solid fa-circle-check"></i> |
+| Comm/Help  | -- | -- | -- |
+| Iterate    | -- | -- | <i class="fa-solid fa-circle-check"></i> |
+| Ext.       | -- | <i class="fa-solid fa-circle-check"></i> | -- |
+| Perf./Scal.| -- | <i class="fa-solid fa-circle-check"></i> |   -- |
+| Our team | -- | -- | <i class="fa-solid fa-circle-check"></i> |
+
+Explanation of the above outcomes:
+
+- Language: I am - and fellow team members are - more proficient in Python than JavaScript and Ruby.
+- Community/Help: This one is not clear cut because I'd need more time with each framework, but first impressions are that HTMX/FastAPI is slightly easier to navigate and get help with - but that is maybe due to just being more familiar with Python vs. the other languages.
+- Iterate: Iterating quickly is essential. Part of that is up to the PM/whoever is running the project - but with respect to technology, having a programming language your comfortable with, and that's easy to write tests for, goes a long way.
+- Extension support: Javascript has the largest ecosystem of any language, but Python is no slouch. Ruby also has a massive number of gems.
+- Performance/scalability: In my playing around with these frameworks Svelte felt the snappiest, and opinions on the internet seem to suggest the same.
+- Good for our team: We have a small team. AFAICT that's unlikely to change - if any, only slightly. Therefore, it's critical that the framework we choose is easy to get started with, easy to maintain, easy to reason about, and easy to debug.
+
+### Squishier factors
+
+There were many other factors that contributed to the outcome that were even less quantifiable:
+
+- Navigation through files: Rails and Sveltekit have larger trees (number of files and nestedness) than HTMX/FastAPI - thus the latter was easier to navigate.
+- Familiarity with the programming language: Mentioned above, but additional things related to the language are:
+  - tests: I'm very comfortable with python testing tools, and very unfamiliar with Ruby and JS testing tools
+  - knowing where to look: this is hard to quantify but for python I just kinda know where to look if I need help whereas with Ruby and Javascript I'm less familiar with where to look
+- What do I like using personally: I love using Ruby, so I'd be happy with landing on Rails. However, Rails wasn't the best option for other reasons
+- What can we hire for easily? In our space, at least AFAICT it'd be easiest to hire folks with a Python background
+- Will the team continue to be the same size? or will it grow/shrink?: AFAICT we're not likely to grow much, so picking a framework that a small team can manage is critical
+- User growth? If we expect rapid growth in number of users of our app, we may want to account for whether the framework can handle the user numb
```

Comment:
Maybe you deleted part of the sentence by accident?

Reply:
thanks will fix